Migrate custom features to AnswerPage answer field rich text editor

virginiacc opened the pull request:
Migrate existing customizations (tip formatter, HTML editor) from the answer rich text field on the Answer model to the analogous field on the AnswerPage.

## Changes

- Remove URL check from `ask_cfpb_tips` and `html_editor` plugins
- Register tip formatter and html editor plugins as features
- Apply tip formatter and html editor features to answer rich text editor on AnswerPage
- Migrate help text from Answer field to AnswerPage field

## Testing

1. Check out branch
2. Run migration
3. Verify that tip formatting and HTML edit view are available on AnswerPage answer field rich text editor and work as expected

## Screenshots
<img width="1221" alt="screen shot 2019-02-19 at 3 23 35 pm" src="https://user-images.githubusercontent.com/778171/53054769-52a8b580-345a-11e9-9f25-0643202b60f7.png">


## Notes

- This should be a short-term fix to allow editors to edit tips and view answer content's HTML. In the longer term, we hope to remove the HTML edit view and handle tip creation via an answer StreamField.

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

higs4281 commented:
This works for me locally and is a great near-term solution for editors.

I noticed one glitch in functionality, at least in Chrome and Firefox. After styling a tip, and following the help_text advice of clicking pencil again to remove it, the `<aside>` tags are not removed. Instead, they double up.

To avoid becoming a blocker for editing, if tip removal code is troublesome, I'd vote to push this PR out for the interim, perhaps with new help_text to advise removing the <aside> tags with the Edit HTML button.

virginiacc commented:
@higginsw Thanks for catching the tip removal issue! I updated the removal code and it seems to be working for me again. I also updated the help text to reflect that the content needs to be re-selected before removal.
